ynput/OpenPype
# Show thumbnails in Workfiles tool

## Changelog Description

- Show thumbnails in Workfiles tool (if there's a `{workfile}_thumbnail.jpg` found next to the workfile).
- Note: Currently only autogenerates thumbnails on save for Maya and Houdini

## Additional info

Draft implementation for [**Thumbnails for Workfiles**](https://community.ynput.io/t/thumbnails-for-workfiles/180/10).

### Tasks

- [x] Make sure there's no overwriting of thumbnails between different file extensions/hosts, so include e.g. .hip in the thumbnail names.
- [ ] Expose settings to enable/disable the workfile thumbnails completely?
- [ ] Expose settings to adjust the thumbnail locations?
- [ ] Thumbnail widget should either hide in "Published" view mode in the workfiles OR show the published workfile thumbnail if it has any.
- [ ] The ThumbnailPainterWidget should likely be moved in the codebase to not need import from the publisher tool.

## Previews

No thumbnail:
![image](https://github.com/ynput/OpenPype/assets/2439881/ebed3219-0178-4e21-bf13-265a6dabdbd0)

With saved thumbnail:
![image](https://github.com/ynput/OpenPype/assets/2439881/9619cd13-a107-431c-9222-aeb043e2ccd0)
![image](https://github.com/ynput/OpenPype/assets/2439881/59793c0d-526c-444f-bcac-ad260f4ac1ac)
![image](https://github.com/ynput/OpenPype/assets/2439881/06ea37de-c033-4812-9ae4-e52ee60144e7)

## Testing notes:

1. Open Maya or Houdini
2. Save scene
3. Open workfiles tool - it should show a thumbnail for the saved scene
